### PR TITLE
2.0 preview

### PR DESCRIPTION
**Pull request by notwaldorf**
Notable changes:
- `bindValue` was marked as deprecated, said to use `value` instead, but `value` wasn't defined and `bindValue` was still doing all the heavy lifting. Also, in 2.0 if you don't declare a property you don't serialize to it either, so anyway, cleaned this up.
- made `value`/`bindValue` be a notifying property vs fire the `foo-changed` event manually. It's currently broken in 2.0 and I'm not 100% sure if we need it anymore, since the long-ago days when `iron-autogrow-textarea` stopped being a type extension.
- cleaned up the `validate` code to match that of `iron-input`, and have the correct behaviour of using the native validity first. I think that's correct and not a giant breaking change

### CODE REVIEW

File: iron-autogrow-textarea.html [hunk 1]
```diff
@@ -127,21 +126,29 @@
...
+       * @type {string|number}
+       */
+      value: {
+        observer: '_valueChanged',
+        type: String,
+        notify: true
+      },
+
+      /**
+       * This property is deprecated, and just mirrors `value`. Use `value` instead.
```
Comment (e111077) [inline iron-autogrow-textarea.html]: Apparently this is borked on FF and Edge. See #100

File: test/index.html [hunk 2]
```diff
@@ -15,10 +15,10 @@
   <body>
     <script>
       WCT.loadSuites([
-        'basic.html',
+        'basic.html?wc-shadydom=true&wc-register=true',
```
Comment (e111077) [inline test/index.html]: wc-ce

File: demo/index.html [hunk 3]
```diff
@@ -18,40 +18,45 @@
     <title>iron-autogrow-textarea demo</title>
 
     <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+
     <link rel="import" href="../../iron-demo-helpers/demo-snippet.html">
     <link rel="import" href="../../iron-demo-helpers/demo-pages-shared-styles.html">
+    <link rel="import" href="../../paper-styles/shadow.html">
```
Comment (e111077) [inline demo/index.html]: is this used?

File: iron-autogrow-textarea.html (hunk 1 above)
Comment (notwaldorf) [inline iron-autogrow-textarea.html]: Fixed in c27bb21

**Comment by notwaldorf** [comment]
@e111077 fixeded all the things!

File: iron-autogrow-textarea.html [hunk 4]
```diff
@@ -74,7 +73,7 @@
         font-family: inherit;
         line-height: inherit;
         text-align: inherit;
-        @apply(--iron-autogrow-textarea);
+        @apply --iron-autogrow-textarea;
       }
 
       ::content textarea:invalid {
```
Comment (bicknellr) [inline iron-autogrow-textarea.html]: `::content`!

**Comment by bicknellr** [comment]
There's also 4 `@apply(...);` in `iron-autogrow-textarea.html`.

File: iron-autogrow-textarea.html (hunk 4 above)
Comment (notwaldorf) [inline iron-autogrow-textarea.html]: 😭

**Pull request merged**